android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt

```diff
@@ -140,9 +146,7 @@ class ModelViewer {
         destroyModel()
         asset = assetLoader.createAssetFromJson(buffer)
         asset?.let { asset ->
-            val resourceLoader = ResourceLoader(engine)
-            resourceLoader.loadResources(asset)
-            resourceLoader.destroy()
+            resourceLoader.asyncBeginLoad(asset)
```

Comment:
This begs for a use of coroutines in Kotlin :)

Reply:
I read up on this and I don't think this language feature is useful to us yet, since most of the interesting work is occurring at a lower level.